gto/git_utils.py

```diff
@@ -0,0 +1,71 @@
+import inspect
+import logging
+import re
+from functools import wraps
+from tempfile import TemporaryDirectory
+from typing import Callable, Dict
+
+from git import Repo
+
+# TODO: make a new type out of Union[str, Repo]
```

Comment:
Do you want to address this in this or next PR?

Reply:
oh! I thought I removed it eventually :)

At first I thought it was a good idea, then reconsidered it. Surely it's a bit redundant to see `Union[str, Repo]` in so many places, but on the other hand for a user it's more understandable than a new type, because he will have to look up for its definition.

So, I'd suggest to remove the TODO, if you agree.

Reply:
done